templates/components/engines/python/agent/tools/interpreter.py

```diff
@@ -72,31 +74,46 @@ def parse_result(self, result) -> List[InterpreterExtraResult]:
 
         try:
             formats = result.formats()
-            base64_data_arr = [result[format] for format in formats]
-
-            for ext, base64_data in zip(formats, base64_data_arr):
-                if ext and base64_data:
-                    result = self.save_to_disk(base64_data, ext)
-                    filename = result["filename"]
-                    output.append(
-                        InterpreterExtraResult(
-                            type=ext, filename=filename, url=self.get_file_url(filename)
+            data_list = [result[format] for format in formats]
+
+            for ext, data in zip(formats, data_list):
+                match ext:
+                    case "png" | "jpeg" | "svg":
+                        result = self.save_to_disk(data, ext)
+                        filename = result["filename"]
+                        output.append(
+                            InterpreterExtraResult(
+                                type=ext,
+                                filename=filename,
+                                url=self.get_file_url(filename),
+                            )
                         )
-                    )
+                        break
+                    case "text" | "html" | "markdown":
+                        output.append(InterpreterExtraResult(type=ext, content=data))
         except Exception as error:
             logger.error("Error when saving data to disk", error)
 
         return output
 
-    def interpret(self, code: str) -> E2BToolOutput:
+    def interpret(self, code: str, file_path: Optional[str] = None) -> E2BToolOutput:
         with CodeInterpreter(api_key=self.api_key) as interpreter:
+            # Upload file to E2B sandbox
```

Comment:
Always upload the file for each code interpreter calling. We can improve this by introducing a chat session feature, then each session can use its own sandbox environment.

Reply:
makes sense. I think this is a general concept we have to add later, e.g. we also need to store the agent's memory in that session store 
as we generate demo code - for now we just have to add an explanation to the user